compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -239,6 +275,18 @@ impl<'context> Elaborator<'context> {
         ident
     }
 
+    pub fn add_existing_variable_to_scope(&mut self, name: String, ident: HirIdent) {
+        let second_span = ident.location.span;
+        let resolver_meta = ResolverMeta { num_times_used: 0, ident, warn_if_unused: true };
```

Comment:
is `num_times_used: 0` accurate? what qualifies as a "use"?

Reply:
A use is a reference within the function, so all definitions start with zero